internal/service/listeners/transfer.go

```diff
@@ -13,28 +15,69 @@ type LogTransfer struct {
 	To   common.Address
 }
 
-func handleIssuerTransferLog(eventLog types.Log) error {
+func (l *Listener) handleIssuerTransferLog(eventLog types.Log) error {
+	l.log.Infof("start handling transfer event")
+
 	var event LogTransfer
 
 	event.From = common.HexToAddress(eventLog.Topics[1].Hex())
 	event.To = common.HexToAddress(eventLog.Topics[2].Hex())
 
-	if event.From.Hex() == "0x0000000000000000000000000000000000000000" {
-		//handle mint
-		fmt.Printf("From: %s\n", event.From.Hex())
-		fmt.Printf("To: %s\n", event.To.Hex())
+	contract, err := l.AddressesQ.FilterByAddresses(l.Address.Hex()).Get()
+	if err != nil {
+		return errors.Wrap(err, "failed to get address from database")
+	}
+
+	if contract == nil {
+		contract, err = l.AddressesQ.Insert(data.Contract{
+			Name:    IssuerContract,
+			Address: l.Address.Hex(),
+		})
+		if err != nil {
+			return errors.Wrap(err, "failed to save new contract")
+		}
+	}
+
+	treeStorage := sql.NewSqlStorage(l.DBConn, contract.Id)
 
-		//TODO: save this in database
-		fmt.Printf("Log Block Number: %d\n", eventLog.BlockNumber)
+	mTree, err := merkletree.NewMerkleTree(l.ctx, treeStorage, 100)
+	if err != nil {
+		return errors.Wrap(err, "failed to get merkle tree")
+	}
+
+	//handle mint
+	if event.From.Hex() == ZeroAddress {
+		err = mTree.Add(l.ctx, event.To.Big(), event.To.Big())
+		if err != nil {
+			return errors.Wrap(err, "failed to add new leaf in merkle tree")
+		}
+
+		err = l.BlocksQ.Upsert(data.Block{
+			ContractAddress: l.Address.Hex(),
+			LastBlockNumber: int64(eventLog.BlockNumber),
+		})
+		if err != nil {
+			return errors.Wrap(err, "failed to save last handled block")
+		}
+
+		l.log.Infof("finish handling mint event")
 		return nil
 	}
 
 	//handle transfer
-	fmt.Printf("From: %s\n", event.From.Hex())
-	fmt.Printf("To: %s\n", event.To.Hex())
+	err = mTree.Delete(l.ctx, event.To.Big())
```

Comment:
You shouldn't just delete it. Check the number of NFTs that the user is owned and if it > 0 decrement it, otherwise delete it